### PR TITLE
Blacklist Hash with VMware Carbon Black EDR from Slack

### PR DESCRIPTION
**Pull request by r7-kszczepanskagorna**
## Requirements

Workflow contributors, verify you have completed the following items by checking them off:

### Checklist
- [ ] Workflow verified as working in a test
- [ ] Workflow bundle includes `Name_of_Workflow.icon` file, `help.md` file, `workflow.spec.yaml` file, and `screenshots` directory
- [ ] Workflow title should follow general [Title of Articles](http://grammar.yourdictionary.com/capitalization/rules-for-capitalization-in-titles.html) format
- [ ] Workflow includes a detailed Description
- [ ] Workflow step names follow [Title of Articles](http://grammar.yourdictionary.com/capitalization/rules-for-capitalization-in-titles.html) format
- [ ] Triggers, Decisions, and other workflow steps include descriptions where needed
- [ ] Workflow includes at least one `Final Report` artifact that summarizes workflow execution details
- [ ] Workflow bundle (directory) is named after the title of the workflow in snake case with first letters capitalized, e.g. `workflows/Name_of_Workflow/`
- [ ] `help.md` file includes Description, Key Features, Requirements, Setup, Technical Details, Version History, and References
- [ ] `workflow.spec.yaml` file includes `products`, `name`, `title`, `description`, `use_cases` tags, `keywords` tags, `source_url`, and the `name` for each screenshot matches a PNG image in the `screenshots` directory

### Assessment

icon-validate
```[*] Validating workflow at .

[*] Running Integration Validators...
[*] Executing validator WorkflowDirectoryNameMatchValidator
[*] Executing validator WorkflowFilesValidator
[*] Executing validator WorkflowHelpValidator
[*] Executing validator WorkflowChangelogValidator
[*] Executing validator WorkflowVendorValidator
[*] Executing validator WorkflowVersionValidator
[*] Executing validator WorkflowExtensionValidator
[*] Executing validator WorkflowSupportValidator
[*] Executing validator WorkflowPNGHashValidator
[*] Executing validator WorkflowICONFileNameValidator
[*] Executing validator WorkflowScreenshotValidator
[*] Executing validator WorkflowTitleValidator
[*] Executing validator WorkflowDescriptionValidator
[*] Executing validator WorkflowNameValidator
[*] Executing validator WorkflowProfanityValidator
[*] Executing validator WorkflowHelpPluginUtilizationValidator
[*] Executing validator WorkflowICONFileValidator
[*] Workflow successfully validated!

----
[*] Total time elapsed: 68.75399999999999ms

```
### Screenshots

Screenshots help your reviewers provide valuable feedback on your contribution. To add a screenshot here, simply drag and drop the PNG files from a file explorer.

#### Workflow Builder

workflow1
workflow2

#### Job
slack1
slack2
slack3

#### Artifact

artifact

```

### CODE REVIEW

File: workflows/Blacklist_Hash_with_VMware_Carbon_Black_EDR_from_Slack/Blacklist_Hash_with_VMware_Carbon_Black_EDR_from_Slack.icon [hunk 1]
```diff
@@ -0,0 +1,1382 @@
...
+        "steps": {
+          "19a25c50-657d-4a7e-bb3d-18a7460eaf16": {
+            "nodeId": "19a25c50-657d-4a7e-bb3d-18a7460eaf16",
+            "name": "Usage",
+            "type": "artifact",
+            "continueOnFailure": false,
+            "isDisabled": false,
+            "parameters": {
+              "input": {
+                "content": "This workflow will blacklist MD5 hashes globally in Carbon Black EDR. To use this workflow provide a blacklist or command in Slack and a hash to perform the command on.\n\nFor example:\n\n* `@Rapid7 InsightConnect blacklist-hash 9de5069c5afe602b2ea0a04b66beb2c0`"
```
Comment (jschipp-r7) [inline workflows/Blacklist_Hash_with_VMware_Carbon_Black_EDR_from_Slack/Blacklist_Hash_with_VMware_Carbon_Black_EDR_from_Slack.icon:231]:
```suggestion
                "content": "This workflow will blacklist MD5 hashes globally in Carbon Black EDR. To use this workflow provide the blacklist-hash command in Slack and an MD5 hash to perform the command on.\n\nFor example:\n\n* `@Rapid7 InsightConnect blacklist-hash 9de5069c5afe602b2ea0a04b66beb2c0`"
```

**Review by jschipp-r7 — CHANGES_REQUESTED** [review]
@r7-kszczepanskagorna for this workflow we shouldn't use Pattern Match because the MD5 can be extracted automatically from the Slack step and the command (i.e. blacklist-hash) is stored in the variable `first_word` of the Slack step - so that can be passed directly to Automated Decision without needing to extract it first with Pattern Match

File: workflows/Blacklist_Hash_with_VMware_Carbon_Black_EDR_from_Slack/Blacklist_Hash_with_VMware_Carbon_Black_EDR_from_Slack.icon [hunk 2]
```diff
@@ -0,0 +1,1140 @@
+{
+  "kom": {
+    "komandVersion": "1.65.2-0-gfd118",
+    "komFileVersion": "2.0.0",
+    "exportedAt": "2020-10-23T15:16:23.757097622Z",
+    "workflowVersions": [
+      {
+        "name": "Blacklist Hash VMware Carbon Black EDR from Slack",
```
Comment (jschipp-r7) [inline workflows/Blacklist_Hash_with_VMware_Carbon_Black_EDR_from_Slack/Blacklist_Hash_with_VMware_Carbon_Black_EDR_from_Slack.icon:8]: Regression in name

```suggestion
        "name": "Blacklist Hash with VMware Carbon Black EDR from Slack",
```

**Review by jschipp-r7 — CHANGES_REQUESTED** [review]
Can a user provide one or more MD5 hashes like the other workflows? If not, let's support that

```
@Rapid7 InsightConnect blacklist-hash 9de5069c5afe602b2ea0a04b66beb2c0 d8e8fca2dc0f896fd7cb4cb0031ba249
```

File: workflows/Blacklist_Hash_with_VMware_Carbon_Black_EDR_from_Slack/Blacklist_Hash_with_VMware_Carbon_Black_EDR_from_Slack.icon [hunk 3]
```diff
@@ -0,0 +1,1268 @@
...
+  "kom": {
+    "komandVersion": "1.66.0-0-g22454",
+    "komFileVersion": "2.0.0",
+    "exportedAt": "2020-10-28T21:39:33.590299128Z",
+    "workflowVersions": [
+      {
+        "name": "Blacklist Hash with VMware Carbon Black EDR from Slack",
+        "type": "runnable",
+        "version": "",
+        "description": "This workflow will blacklist a MD5 hash globally in Carbon Black EDR from a Slack command.",
```
Comment (mrinehart-r7) [inline workflows/Blacklist_Hash_with_VMware_Carbon_Black_EDR_from_Slack/Blacklist_Hash_with_VMware_Carbon_Black_EDR_from_Slack.icon:11]:
```suggestion
        "description": "This workflow will blacklist an MD5 hash globally in VMware Carbon Black EDR from a Slack command.",
```

File: workflows/Blacklist_Hash_with_VMware_Carbon_Black_EDR_from_Slack/Blacklist_Hash_with_VMware_Carbon_Black_EDR_from_Slack.icon [hunk 4]
```diff
@@ -0,0 +1,1268 @@
...
+          "6bf1d4a9-cc6c-4991-8a1d-e4efbb763efc": {
+            "nodeId": "6bf1d4a9-cc6c-4991-8a1d-e4efbb763efc",
+            "name": "Post Request Received",
+            "type": "action_chatops",
+            "continueOnFailure": false,
+            "isDisabled": false,
+            "parameters": {
+              "input": {
+                "attachments": [],
+                "message": "🚀 Request received! Running Carbon Black EDR workflow... 🚀",
```
Comment (mrinehart-r7) [inline workflows/Blacklist_Hash_with_VMware_Carbon_Black_EDR_from_Slack/Blacklist_Hash_with_VMware_Carbon_Black_EDR_from_Slack.icon:368]:
```suggestion
                "message": "🚀 Request received! Running VMware Carbon Black EDR workflow... 🚀",
```

File: workflows/Blacklist_Hash_with_VMware_Carbon_Black_EDR_from_Slack/Blacklist_Hash_with_VMware_Carbon_Black_EDR_from_Slack.icon [hunk 5]
```diff
@@ -0,0 +1,1268 @@
...
+          },
+          "8fdf05b7-77e7-4058-9488-94052d45dace": {
+            "nodeId": "8fdf05b7-77e7-4058-9488-94052d45dace",
+            "name": "Blacklist Success",
+            "type": "artifact",
+            "continueOnFailure": false,
+            "isDisabled": false,
+            "parameters": {
+              "input": {
+                "content": ":rotating_light: Hash action successfully executed for `{{[cbfdb9ff-28e2-4e9e-a643-7605ca1c279e].[$item]}}` :rotating_light:"
```
Comment (mrinehart-r7) [inline workflows/Blacklist_Hash_with_VMware_Carbon_Black_EDR_from_Slack/Blacklist_Hash_with_VMware_Carbon_Black_EDR_from_Slack.icon:566]:
```suggestion
                "content": ":rotating_light: Blacklist hash action successfully executed for `{{[cbfdb9ff-28e2-4e9e-a643-7605ca1c279e].[$item]}}` :rotating_light:"
```
Just a little more clarification to the user

File: workflows/Blacklist_Hash_with_VMware_Carbon_Black_EDR_from_Slack/Blacklist_Hash_with_VMware_Carbon_Black_EDR_from_Slack.icon [hunk 6]
```diff
@@ -0,0 +1,1268 @@
...
+          },
+          "ee41ed11-ed0f-4596-8097-c3ce2ff03e9b": {
+            "nodeId": "ee41ed11-ed0f-4596-8097-c3ce2ff03e9b",
+            "name": "Incorrect Hash Result",
+            "type": "artifact",
+            "continueOnFailure": false,
+            "isDisabled": false,
+            "parameters": {
+              "input": {
+                "content": "Carbon Black EDR requires a MD5 hash but it was not provided. Please run this workflow again with a MD5 hash.\n\nFor example:\n\n* `@Rapid7 InsightConnect blacklist-hash 9de5069c5afe602b2ea0a04b66beb2c0`\n"
```
Comment (mrinehart-r7) [inline workflows/Blacklist_Hash_with_VMware_Carbon_Black_EDR_from_Slack/Blacklist_Hash_with_VMware_Carbon_Black_EDR_from_Slack.icon:1005]:
```suggestion
                "content": "VMware Carbon Black EDR requires an MD5 hash but one was not provided. Please run this workflow again with an MD5 hash.\n\nFor example:\n\n* `@Rapid7 InsightConnect blacklist-hash 9de5069c5afe602b2ea0a04b66beb2c0`\n"
```

File: workflows/Blacklist_Hash_with_VMware_Carbon_Black_EDR_from_Slack/help.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
+# Description
+
+This workflow blacklists a MD5 hashes globally with VMware Carbon Black EDR via Slack command and reports information back to Slack.
```
Comment (mrinehart-r7) [inline workflows/Blacklist_Hash_with_VMware_Carbon_Black_EDR_from_Slack/help.md:3]:
```suggestion
This workflow blacklists MD5 hashes globally with VMware Carbon Black EDR via Slack command and reports information back to Slack.
```

**Review by mrinehart-r7 — CHANGES_REQUESTED** [review]

**Review by mrinehart-r7 — APPROVED** [review]

**Pull request merged**